src/core/app-template.cc

```diff
@@ -109,12 +100,33 @@ app_template::app_template(app_template::config cfg)
 {
 }
 
+void
+app_template::add_extra_options(boost::program_options::options_description& opts) {
+    opts.add_options()
+            ("help,h", "show help message")
+            ;
+    opts.add_options()
+            ("help-seastar", "show help message about seastar options")
+            ;
+    opts.add_options()
+            ("help-loggers", "print a list of logger names and exit")
+            ;
+}
+
 app_template::~app_template() = default;
 
 const app_template::seastar_options& app_template::options() const {
     return _opts;
 }
 
+void
+app_template::set_app_visible_options_for_help(boost::program_options::options_description* opts) {
+    _app_visible_opts_for_help = opts;
+    if (_app_visible_opts_for_help) {
+        add_extra_options(*_app_visible_opts_for_help);
+    }
+}
```

Comment:
could be more explicit, like:
```c++
if (opts) {
    add_extra_options(*opts);
    _app_visible_opts_for_help = opts;
}
```

this way, it's clear that we mutate the pointee of `opts`, and it's a noop, if `opts` is  `nullptr`. but i'd suggest accept a reference instead. like
```c++
set_app_visible_options_for_help(boost::program_options::options_description& opts)
```
as, why would the caller want to pass a nullptr for a noop?